trading/views.py

```diff
@@ -0,0 +1,98 @@
+from rest_framework import status, mixins, viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from trading.models import Offer, Trade
+from user.permissions import (
+    BlackListPermission,
+    IsOwner,
+    CreateOfferPermission
+)
+from trading.serializers import (
+    OfferRetrieveSerializer,
+    OfferCreateSerializer,
+    OfferListSerializer,
+    TradeListSerializer,
+    TradeRetrieveSerializer
+)
+from trading.filters import OfferUserFilter
+
+
+class OfferViewSet(mixins.ListModelMixin,
+                   mixins.CreateModelMixin,
+                   mixins.RetrieveModelMixin,
+                   viewsets.GenericViewSet):
+    """A viewset for model Offer that provides `retrieve`,
+     `create`, and `list` actions with specific permissions by action."""
+    model = Offer
+    queryset = Offer.objects.all()
+    filterset_class = OfferUserFilter
+
+    serializer_action_classes = {
+        'list': OfferListSerializer,
+        'retrieve': OfferRetrieveSerializer,
+        'create': OfferCreateSerializer
+    }
+
+    permission_classes_by_action = {'create': (
+                                        IsAuthenticated,
+                                        CreateOfferPermission,
+                                    ),
+                                    'list': (IsAuthenticated,),
+                                    'retrieve': (
+                                        IsAuthenticated,
+                                        BlackListPermission,
+                                        IsOwner
+                                    ),
+                                    }
+
+    def get_serializer_class(self):
+        return self.serializer_action_classes[self.action]
+
+    def get_permissions(self):
+        try:
+            return [permission() for permission
+                    in self.permission_classes_by_action[self.action]]
+        except KeyError:
+            return [permission() for permission in self.permission_classes]
```

Comment:
Error handling takes a long time. So i think there will be great another option.
```python
permissions = self.permission_classes_by_action.get(self.action, self.permission_classes)
return [permission() for permission in permissions]
```